frontend/src/composables/String.js

```diff
@@ -51,3 +51,44 @@ export const generateUuid = (length = 6) => {
         byte.toString(36).padStart(2, '0')
     ).join('').substring(0, length)
 }
+
+/**
+ * Generates a hash string from a given input string.
+ * The hash value is computed using the FNV-1a hashing algorithm.
+ * The length of the returned hash can be configured.
+ *
+ * @param {string} str - The input string to be hashed.
+ * @param {number} [length=5] - The desired length of the resulting hash string. Defaults to 5 characters.
+ * @returns {string} - The computed hash string truncated to the specified length.
+ */
+export const hashString = (str, length = 5) => {
+    let h = 2166136261 >>> 0
+    for (let i = 0; i < str.length; i++) {
+        h ^= str.charCodeAt(i)
+        h = Math.imul(h, 16777619)
+    }
+    return (h >>> 0).toString(16).substring(0, length)
+}
```

Comment:
Nice fnv1a hash however I am a bit confused why the length limit to 5 chars and not leave it to the default 8 chars an FNV returns (effective hash space for 8 chars ~32bit, 5 chars = ~20 bits) so in effect increasing the chance of collision for no apparent reason (i.e. vue keys the keys dont need to be short - we could just keep the full 8 hex chars!?!)